packages/react-components/virtualizer/src/Virtualizer.types.ts

```diff
@@ -0,0 +1,104 @@
+import type { ReactNode, MutableRefObject } from 'react';
+import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+
+export type VirtualizerSlots = {
+  before?: Slot<'div', 'td'>;
+  beforeContainer?: Slot<'div', 'tr'>;
+  after?: Slot<'div', 'td'>;
+  afterContainer?: Slot<'div', 'tr'>;
+};
+
+export type VirtualizerState = ComponentState<VirtualizerSlots> & {
+  /**
+   * The current virtualized array of children to show in the DOM.
+   */
+  virtualizedChildren: ReactNode[];
+  /**
+   * The current start index for the virtualizer, all previous index's will be removed from DOM.
+   */
+  virtualizerStartIndex: number;
+  /**
+   * Current buffer height required at beginning of array.
+   */
+  afterBufferHeight: number;
+  /**
+   * Current buffer height required at end of array.
+   */
+  beforeBufferHeight: number;
+  /**
+   * The total current height of the scrollBiew/child content.
+   */
+  totalVirtualizerHeight: number;
+  /**
+   * The scroll direction (vertical vs horizontal) - defaults to Vertical if not provided
+   */
+  isHorizontal: boolean;
```

Comment:
How would RTL support be done with this? I took the `Horizontal` example and added `dir="rtl"` and was hit with unexpected bugs
![msedge_rFM6ZVMD72](https://user-images.githubusercontent.com/20744592/206183218-0e26c4dd-fa64-40d7-94e9-cbe04fdecd98.gif)

Reply:
I believe for this one we'd need 'isReversed' set to true, I'll take a look.

Reply:
Confirmed, isReversed flag should be enabled for RTL.

Reply:
Agreed! Ran into a hiccup here with useFluent however, it seems to only output 'ltr' in this example, even with display: 'rtl' visible in css. Does the hook check parent containers if the 'rtl' is external to the component?

Reply:
Right now, this component is mostly intended to be a 'raw' version of a virtualizer and is completely detached from the scrollview, so we don't have a container internal to the component that 'rtl' would be placed on.

Next steps are likely to build out a more off the shelf version of this class that also wrap a scrollView and automatically handle sizing the virtualizer length etc. This might be a good place for the RTL functionality to be automated as part of a simplified List wrapper experience.

Reply:
I agree handling RTL internally would help consumers. Often this is something that consumers forget about so helping them out would be great. However, I also think ensuring it is handled without having a consumer explicitly set rtl in a Fluent provider or hook would be more ideal. Looking up the tree should be possible. In previous Fluent versions I saw inconsistent usage in terms of setting RTL in <Fluent/Fabric> and it also doesn't handle bidi/mixed scenarios well. Leveraging what's in the actual DOM helps solve that better.

Reply:
One question I do have is if we consider RTL core. I think we could argue that it is. Are there any virtualization scenarios in which that shouldn't be handled? If so, and we built it in, are there good escape hatches? For the most part I think we want RTL handled properly always so there's an argument to be made to build that piece in. Scrolling I think would be an extension/wrap scenario though as there are plenty of places I could think of where one wouldn't want a scroll view baked in or only a partial UI virtualized which this allows for.